### PR TITLE
remove all uses of WorkflowAction.configure()

### PR DESCRIPTION
**Pull request by anew**

### CODE REVIEW

**Comment by anew** [comment]
Build is running here: http://builds.cask.co/browse/CDAP-RBT448-1

**Comment by gokulavasan** [comment]
Do we want to leave at least one test which uses the old configure API so that we know it is working? We can convert that too when we remove the old API?

**Comment by gokulavasan** [comment]
One comment. LGTM otherwise.

**Comment by anew** [comment]
I added a custom action that implements the legacy configure(). It turns out that even before my change, that did not work, because it does not set the class name in the action spec. It works if you extend AbstractWorkflowAction, but not if you implement WorkflowAction directly. I will try to fix this, too.

**Comment by anew** [comment]
@gokulavasan I fixed all the issues I could find locally on my laptop. New build is running here: http://builds.cask.co/browse/CDAP-RBT448. Please review again. Thanks

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/workflow/WorkflowNodeCreator.java [hunk 1]
```diff
@@ -56,11 +57,9 @@ static WorkflowNode createWorkflowCustomActionNode(WorkflowAction action) {
     Preconditions.checkArgument(action != null, "WorkflowAction is null.");
     WorkflowActionSpecification spec;
     if (action instanceof AbstractWorkflowAction) {
-      DefaultWorkflowActionConfigurer configurer = new DefaultWorkflowActionConfigurer(action);
-      ((AbstractWorkflowAction) action).configure(configurer);
-      spec = configurer.createSpecification();
+      spec = DefaultWorkflowActionConfigurer.configureAction((AbstractWorkflowAction) action);
```
Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/workflow/WorkflowNodeCreator.java]: i guess we don't need this casting any more?

Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/workflow/WorkflowNodeCreator.java]: do we still need that casting?

File: cdap-app-fabric/src/test/java/co/cask/cdap/AppWithWorkflow.java [hunk 2]
```diff
@@ -89,11 +90,9 @@ public DummyAction(String name) {
...
 
     @Override
-    public WorkflowActionSpecification configure() {
-      return WorkflowActionSpecification.Builder.with()
-        .setName(name)
-        .setDescription(getDescription())
-        .build();
+    public void configure(WorkflowActionConfigurer configurer) {
+      super.configure(configurer);
+      setName(name);
```
Comment (gokulavasan) [inline cdap-app-fabric/src/test/java/co/cask/cdap/AppWithWorkflow.java]: any reason to exclude the setDescription method?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/workflow/WorkflowNodeCreator.java (hunk 1 above)
Comment (anew) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/workflow/WorkflowNodeCreator.java]: We do need it. The configureAction() requires an AbstractWorkflowAction.

File: cdap-app-fabric/src/test/java/co/cask/cdap/AppWithWorkflow.java (hunk 2 above)
Comment (anew) [inline cdap-app-fabric/src/test/java/co/cask/cdap/AppWithWorkflow.java]: because getDescription is deprecated, and defaults to the same as name.

**Comment by gokulavasan** [comment]
Minor nit. LGTM :+1:

**Pull request merged**